### PR TITLE
Add telemetry CLI flag section

### PR DESCRIPTION
**Pull request by antekresic**
## Description

The introduction of this new section will reduce confusion between the telemetry
configuration for the telemetry that Promscale produces and various other types
of telemetry Promscale is used to store and query.

## Merge requirements

Please take into account the following non-code changes that you may need to make with your PR:

- [x] CHANGELOG entry for user-facing changes (_not needed, changelog already includes CLI renaming changes_)
- [x] Updated the relevant documentation

### CODE REVIEW

**Review by Harkishen-Singh — APPROVED** [review]

**Comment by JamesGuthrie** [comment]
I'm not sure that I understand the motivation behind this change. I'm also not sure what it clears up exactly. The term `telemetry` seems to have three different meanings at the moment:

1. Tracking/sending promscale usage data (usage telemetry)
2. Promscale's telemetry (logs, metrics and traces that it emits about itself)
3. End-user telemetry (logs, metrics and traces that Promscale stores, the reason it's used).

**Comment by jgpruitt** [comment]
I tend to agree with James. Are these flags related to the logs that the connector spits out about its own activity?

File: docs/cli.md [hunk 1]
```diff
@@ -48,13 +48,13 @@ You can also find information on flags with `promscale_<version> -help`.
 | db.uri | string | | TimescaleDB/Vanilla PostgresSQL URI. Example:`postgres://postgres:password@localhost:5432/timescale?sslmode=require` |
 | db.user | string | postgres | TimescaleDB/Vanilla Postgres user. |
 
-## Logging flags
+## Telemetry flags
```
Comment (cevian) [inline docs/cli.md:51]:
```suggestion
## Telemetry flags (for telemetry generated by the Promscale connector itself)
```

**Review by cevian — APPROVED** [review]

**Comment by antekresic** [comment]
This change is about the Promscale telemetry it emits about itself. I understand there are multiple tracks of telemetry when talking about Promscale, maybe we can figure out a better name for this group.

I'm open to suggestions for naming, the main thing that we separate out these options from the other telemetry which will be ingested into Promscale.

**Review by jgpruitt — APPROVED** [review]

**Pull request merged**